.github/workflows/bot.yml

```diff
@@ -156,3 +156,33 @@ jobs:
         uses: stefanzweifel/git-auto-commit-action@v5
         with:
           commit_message: "style: apply lint fixes"
+
+  Cache_Benchmarks:
+    name: Caching Criterion benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Install cargo-criterion
+        run: cargo install cargo-criterion
+
+      - name: Run Criterion Benchmarks
+        run: cargo criterion --message-format=json > benches/main_benchmarks.json
```

Comment:
I think install, run and print can be combine to one step. In the the executed results should be cached. Printing should be more human readable, compressed json isn't very readable.